surfstudio/NodeKit
# Fix memory leaks

# What is done

- Add weak modifiers for some properties that captures in closure 

# Note 
AborterNode is not used by default in any chain, so I didn't test it 

# How to check

- Launch any project with nodeKit request from master(3.4.1 version) and check leaks with Xcode Instruments 
- Launch same project from fix_leaks branch and again check leaks

Below example from different branches. First(picture) from fix_leaks branch, second(video) from base branch 

<img width="1425" alt="Screenshot 2023-05-02 at 11 03 17" src="https://user-images.githubusercontent.com/20303349/235602789-13b23cea-0b84-4ce4-ba8d-427f138792aa.png">
 
 
https://user-images.githubusercontent.com/20303349/235603005-9eedf4a0-20a6-4458-a0d4-1ea7275e4f55.mov
